lib/AstOfLlbc.ml

```diff
@@ -782,17 +778,26 @@ let build_trait_clause_mapping env (trait_clauses: C.trait_clause list) =
       List.map (fun (item_name, decl_id) ->
         let decl = env.get_nth_function decl_id in
         let ts = { K.n = List.length trait_decl.generics.types; n_cgs = List.length trait_decl.generics.const_generics } in
-        (clause_id, item_name), (clause_generics, ts, trait_decl.C.name, decl.C.signature)
+        (C.Clause clause_id, item_name), (clause_generics, ts, trait_decl.C.name, decl.C.signature)
         ) trait_decl.C.required_methods @
       List.map (fun (item_name, decl_id) ->
         match decl_id with
         | Some decl_id ->
             let decl = env.get_nth_function decl_id in
             let ts = { K.n = List.length trait_decl.generics.types; n_cgs = List.length trait_decl.generics.const_generics } in
-            (clause_id, item_name), (clause_generics, ts, trait_decl.C.name, decl.C.signature)
+            (C.Clause clause_id, item_name), (clause_generics, ts, trait_decl.C.name, decl.C.signature)
         | None ->
             failwith ("TODO: handle provided trait methods, like " ^ item_name)
-            ) trait_decl.C.provided_methods
+      ) trait_decl.C.provided_methods @
+      List.flatten (List.mapi (fun i (parent_clause: C.trait_clause) ->
+        (* Mapping of the methods of the parent clause *)
+        let m = build_trait_clause_mapping env [ parent_clause ] in
+        List.map (fun ((clause_id, m), v) ->
+          (* This is the parent clause `i` of `clause_id` -- see comments in charon/types.rs  *)
+          let id = C.(ParentClause (clause_id, trait_id, TraitClauseId.of_int i)) in
```

Comment:
slightly strange that I have to rebuild this myself, please check that I'm doing the right thing

Reply:
well I'm recursing into the parent clause and it has its own clause id relative to itself, without the ParentClause around it